src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -198,6 +198,12 @@ private TypeSpec CreateTypeSpec(TypeParseInfo typeParseInfo)
                 else if (IsCollection(type))
                 {
                     spec = CreateCollectionSpec(typeParseInfo);
+
+                    // fallback to treating as an object, if we can create it
+                    if (spec is UnsupportedTypeSpec && type is INamedTypeSymbol && !type.IsAbstract)
```

Comment:
What types of inputs are deemed to be collections but then return `UnsupportedTypeSpec`? Could these checks be integrated in the `IsCollection` predicate instead?

Reply:
I considered that  -- it's all of the checks here: https://github.com/dotnet/runtime/blob/dcf9814b9194cd1ab14ed8108098f06e0fd53e67/src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs#L363-L540

IMO pulling those out into the checking method would create two codepaths we have to keep in sync, since many of these checks would need to be duplicated when examining the type and deciding how it would bind.

Reply:
Could they be merged into a single `Try-` method that just `out`s a valid model? Just a suggestion, I realize we have a deadline today and time is short.

Reply:
It's actually a bit more nuanced than just returning false here.  Much of the time we want the diagnostics info from this.  The fallback turns out to be a bit simpler than this.

It's if we didn't find a way to bind as a collection, and the type can be created and does not implement ICollection<> nor IDictionary<,> than treat as an object.